tools/pv_tree_generator/build_tree.py

```diff
@@ -104,9 +107,59 @@ def json_blob(self):
             result['t'] = 'p'
         if result['cd']:
             result['cd'] = text_format.filter_and_sort(self.name, result['cd'])
+        if result['t'] == 'v':
+            result['d'] = find_denominators(self.pos, cpvs, self.stats_vars_all,
+                                            result['sv'])
+
         return result
 
 
+def powerset(iterable: List) -> Iterator:
+    """Returns all subsets of a set.
+
+    powerset([1,2,3]) --> () (1,) (2,) (3,) (1,2) (1,3) (2,3) (1,2,3)
+    https://docs.python.org/3/library/itertools.html#itertools-recipes
+    """
+    s = list(iterable)
+    return chain.from_iterable(combinations(s, r) for r in range(len(s) + 1))
+
+
+def find_denominators(pop_obs: PopObsSpec, cpvs: Dict[str, str],
+                      stats_vars_all: Dict[str, List[StatsVar]],
+                      stats_vars_to_exclude: List[str]) -> List[str]:
+    """Finds possible StatVars that can be used as per capita denominators for
+    a StatVar.
+
+    The dpvs of "pop_obs" and "cpvs" are combined to form the final pv mapping.
+    A denominator is included if its pvs are a subset of the final pv mapping.
+    E.g., if dpvs is {'age': '15YearsOnwards'} and cpvs is {'gender': 'female'},
+    the denominators are 'Count_Person', 'Count_Person_Female',
+    'Count_Person_15OrMoreYears', and 'Count_Person_Female_15OrMoreYears'.
```

Comment:
Can we order the sv by the importance, ie, 
- the sv corresponding to the dpv population is at the beginning.
- the sv corresponding to 0 pv population: Count_Person, Count_Household comes next,
- then all the other population that corresponding to the cpvs.

Reply:
done